eth/handler_eth.go

```diff
@@ -64,8 +67,9 @@ func (h *ethHandler) PeerInfo(id enode.ID) interface{} {
 
 // AcceptTxs retrieves whether transaction processing is enabled on the node
 // or if inbound transactions should simply be dropped.
-func (h *ethHandler) AcceptTxs() bool {
-	if h.noTxGossip {
+func (h *ethHandler) AcceptTxs(peer *eth.Peer) bool {
+	// Check if peer is allowed for transaction gossip
+	if !peer.IsAllowedForTxGossip() {
```

Comment:
See comment above, then you can here just do

```suggestion
	if !h.txGossipAllowed(peer) {
```